src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -567,6 +569,23 @@ private async Task PackageSourcesChangedAsync(IReadOnlyCollection<PackageSourceC
             }
         }
 
+        private async Task IsCentralPackageManagementEnabledAsync(CancellationToken cancellationToken)
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async delegate
+            {
+                // Go off the UI thread to perform non-UI operations
+                await TaskScheduler.Default;
+
+                if (!Model.IsSolution)
+                {
+                    foreach (IProjectContextInfo project in Model.Context.Projects)
+                    {
```

Comment:
I don't understand this loop... so the last project wins? It looks like it's overriding a shared member variable.

Reply:
There is only 1 project because is not a Solution, I prefer using a foreach here to be clear that Model.Context.Projects is an array but since is not a solution it can only have 1 element (or maybe empty?).

Reply:
[In other places](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#LL293C27-L293C58), we're not afraid to make this assertion: `Model.Context.Projects.First();`

I'd say it's more clear than a loop that should always be 1 item. If it ever were more than 1 item, I'd rather an exception occur rather than "last project wins".

Reply:
In fact, not sure why we aren't using `Model.Context.Projects.Single();` in both places 🤔